src/main/java/hudson/remoting/RemoteClassLoader.java

```diff
@@ -105,7 +105,14 @@
      */
     private final Map<String,ClassReference> prefetchedClasses = Collections.synchronizedMap(new HashMap<String,ClassReference>());
 
-    public static ClassLoader create(ClassLoader parent, IClassLoader proxy) {
+    /**
+     * Creates a remotable classloader
+     * @param parent Parent classloader. Can be {@code null} if there is no delegating classloader
```

Comment:
This null goes to the JVM internals. Javadoc for ClassLoader#parent:

```
// The parent class loader for delegation
    // Note: VM hardcoded the offset of this field, thus all new fields
    // must be added *after* it.
```

In the implementation of the `ClassLoader` I see the execution is being passed to the BootStrap Classloader then, e.g.:

```java
  protected Class<?> loadClass(String name, boolean resolve)
        throws ClassNotFoundException
    {
        synchronized (getClassLoadingLock(name)) {
            // First, check if the class has already been loaded
            Class<?> c = findLoadedClass(name);
            if (c == null) {
                long t0 = System.nanoTime();
                try {
                    if (parent != null) {
                        c = parent.loadClass(name, false);
                    } else {
                        c = findBootstrapClassOrNull(name);
                    }
                } catch (ClassNotFoundException e) {
                    // ClassNotFoundException thrown if class not found
                    // from the non-null parent class loader
                }
...
```